lib/elixir/lib/task/supervised.ex

```diff
@@ -10,26 +12,37 @@ defmodule Task.Supervised do
   end
 
   def start_link(caller, link, info, fun) do
-    :proc_lib.start_link(__MODULE__, :reply, [caller, link, info, fun])
+    {:ok, spawn_link(caller, link, info, fun)}
   end
 
-  def async(caller, info, mfa) do
-    initial_call(mfa)
-    ref = receive do: ({^caller, ref} -> ref)
-    send caller, {ref, do_apply(info, mfa)}
+  def spawn_link(caller, link \\ :nolink, info, fun) do
+    :proc_lib.spawn_link(__MODULE__, :reply, [caller, link, info, fun])
```

Comment:
Last question: do we even need to call proc_lib at all? At this moment, it is going three things:

1. setting the initial call, which we override
2. setting the ancestors, which is easy to replicate
3. handling errors, which we have some code to handle errors somewhat differently

It looks like we are not really gaining anything by using proc_lib, unless you think that we are using most of proc_lib error handling instead of working around it.